### ISSUE TITLE
Dealing with spatial polygons (species distribution)

### ISSUE THREAD

**Issue by ahasverus** [FRBCesab/funbiogeo#5]
Currently we consider species distribution at point level. It could be insteresting to develop the package to take into account species distribution at large polygons level (i.e. IUCN spatial polygon, etc.).

**Comment by Rekyt** [comment]
That was one interest on working with `sf` objects over regular data.frame with coordinates columns. Because then the package was agnostic about the type of spatial data used for plots.

**Comment by ahasverus** [comment]
I agree. But I think we need to adapt our workflow. When dealing with large spatial polygons (i.e. IUCN) polygons do not correspond to sites (they can overlap). So to compute the Pcover (and other metrics) we need to create sites from the raster. Right? Could we not use the same logic for points (or small polygons like plots)? First create new sites (grid cells centroids) then compute all metrics.

**Comment by Rekyt** [comment]
Hum... Then we're thinking about different kind of sites objects.

When you said polygons I thought about regular polygons that would define the sampled locations of sites.
While, if I understood well, you're talking about stacked polygons of individual species ranges.... Which in this case, make no sense to compute `extract()` environmental data on.

We could assume the data is either point data or polygons from a regular grid representing assemblages. And for people that use individual species polygons suggest in the documentation to transform them into assemblages along a grid.
